index.d.ts

```diff
@@ -18,45 +24,54 @@ declare module 'connected-react-router' {
   }
 
   export interface RouterState {
-    location: Location
-    action: RouterActionType.POP | RouterActionType.PUSH
+    location: Location;
+    action: RouterActionType.POP | RouterActionType.PUSH;
+  }
+
+  export const LOCATION_CHANGE: "@@router/LOCATION_CHANGE";
+  export const CALL_HISTORY_METHOD: "@@router/CALL_HISTORY_METHOD";
+
+  export interface LocationChangeAction {
+    type: typeof LOCATION_CHANGE;
+    payload: RouterState;
+  }
+
+  export interface CallHistoryMethodAction {
+    type: typeof CALL_HISTORY_METHOD;
+    payload: LocationActionPayload;
   }
 
-  export const LOCATION_CHANGE: '@@router/LOCATION_CHANGE'
-  export const CALL_HISTORY_METHOD: string
+  export type RouterAction = LocationChangeAction | CallHistoryMethodAction;
```

Comment:
Typescript n00b question: Is it backward compatible with `export interface RouterAction` which was replaced by this `export type RouterAction` alias of interfaces?

Reply:
It should work the exact same way, the only thing an interface really gives you is the ability to extend that type more, which these action types will never need to be extended since they are the literal actions that get passed into the reducer.